src/dsp_tools/commands/ingest_xmlupload/ingest_files/ingest_files.py

```diff
@@ -0,0 +1,44 @@
+from datetime import datetime
+from pathlib import Path
+from time import sleep
+
+from loguru import logger
+
+from dsp_tools.cli.args import ServerCredentials
+from dsp_tools.commands.ingest_xmlupload.bulk_ingest_client import BulkIngestClient
+from dsp_tools.utils.connection import Connection
+from dsp_tools.utils.connection_live import ConnectionLive
+
+
+def ingest_files(creds: ServerCredentials, shortcode: str) -> bool:
+    """
+    Kick off the ingest process on the server, and wait until it has finished.
+    Then, retrieve the mapping CSV from the server and save it in the CWD.
+
+    Args:
+        creds: credentials to log in on the server
+        shortcode: shortcode of the project
+
+    Returns:
+        success status
+    """
+    con: Connection = ConnectionLive(creds.server)
+    con.login(creds.user, creds.password)
+    bulk_ingest_client = BulkIngestClient(creds.dsp_ingest_url, con.get_token(), shortcode)
+    bulk_ingest_client.kick_off_ingest()
+    sleep(5)
+
+    while not (mapping := bulk_ingest_client.retrieve_mapping()):
+        sleep(10)
```

Comment:
Assuming that the bulk ingest is done with a lot of files should we maybe make this longer, so that we don't spam the server?

Reply:
I think 10s is easily justifiable, refusing the mapping is a very cheap operation on the server